src/main/java/com/klarna/hiverunner/HiveServerContainer.java

```diff
@@ -149,6 +154,52 @@ public String apply(@Nullable Object[] objects) {
         }
     }
 
+    /*
+    There's an obscure bug in HiveRunner where if you have a View query with a Join, it will fail to execute the query
+    if the table names after the join are not lowercase, so this method transforms the query so that the parts
+    of the query that are case insensitive are lowercase
+    */
+    public String transformQuery(String hiveql) {
+        String result = hiveql;
+        int indexOn = hiveql.toLowerCase().indexOf(" on ");
+        if(hiveql.contains("'")){
+            //First we leave the query untouched from whatever is before the on
+            result = hiveql.substring(0, indexOn);
+            int indexFirstComa = hiveql.indexOf("'");
+            String subString = hiveql.substring(indexOn, hiveql.length());
+            //Count how many case sensitive strings there is
+            int count = StringUtils.countMatches(subString, "'");
+            int indexStart = subString.indexOf("'");
+
+            for(int i = 0; i < count/2; i++) {
+                indexStart = subString.indexOf("'");
+                //First we turn to lowercase whatever is before the case sensitive string
+                result = result + subString.substring(0,indexStart).toLowerCase();
+                //We add the case sensitive part without lowercasing it
+                String caseSensitiveString = subString.substring(indexStart+1, subString.length());
+                int indexEnd = caseSensitiveString.indexOf("'");
+                caseSensitiveString = caseSensitiveString.substring(0, indexEnd);
+                result = result +"'"+ caseSensitiveString +"'";
+                subString = subString.substring(indexStart+caseSensitiveString.length()+2,subString.length());
```

Comment:
I'm not sure if its just me, but I'm finding this method fairly confusing, wondering if it can be simplified. 
Also I dont think it works if you have some string which comes after the last quote, I ran it and got 
```
before: On tablEA.id = TAbleb.id WHERE tablea.value ='V2:MiXedCases' And tABLeA.x = 2;
after: on tablea.id = tableb.id where tablea.value ='V2:MiXedCases' And tABLeA.x = 2;
```